spdmlib/src/crypto/x509v3.rs

```diff
@@ -381,18 +381,26 @@ fn get_key_usage_value(data: &[u8]) -> SpdmResult<(bool, u8)> {
     let mut find_key_usage = false;
     let len = data.len();
     let key_usage_oid_len = OID_KEY_USAGE.len();
-    if len < 1 {
+    if len < 1 || data[0] != ASN1_TAG_SEQUENCE {
         return Err(SPDM_STATUS_VERIF_FAIL);
     }
     let (data_length, bytes_consumed) = check_length(&data[1..])?;
     if len < 1 + data_length + bytes_consumed {
         Err(SPDM_STATUS_VERIF_FAIL)
     } else {
         let mut index = 1 + bytes_consumed;
-        while index < data_length {
+        // search KEY_Usage OID in Extensions Sequence
+        while index < len {
+            if index + 1 >= len {
+                return Err(SPDM_STATUS_VERIF_FAIL);
+            }
             let (payload_length, bytes_consumed) = check_length(&data[index + 1..])?;
+            if index + 1 + bytes_consumed + payload_length > len {
```

Comment:
to avoid overflow, the best way is to use minus. `if bytes_consumed + payload_length > len - (index + 1)`